### PR TITLE
Add license definitions to arxiv-base and simple test

### PR DESCRIPTION
**Pull request by mhl10**
This is mostly a conversion of the essentials from `arxiv-lib:arXiv::Config::Licenses`.

### CODE REVIEW

File: arxiv/license.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    'http://creativecommons.org/licenses/by/4.0/',
+    'http://creativecommons.org/licenses/by-nc-sa/3.0/':
+    'http://creativecommons.org/licenses/by-nc-sa/4.0/',
+    'http://creativecommons.org/licenses/publicdomain/':
+    'http://creativecommons.org/publicdomain/zero/1.0/'
+}
+
+CURRENT_LICENSES = {
+    k: v for k, v in LICENSES.items()
+    if all(k in v for k in ('is_current', 'order')) and v['is_current']
```
Comment (erickpeirson) [inline arxiv/license.py]: This is totally nit-picky, but any reason not to make this more brutish for the sake of clarity? eg.

```if 'is_current' in v and 'order' in v and v['is_current']```

File: arxiv/license.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+"""arXiv license definitions."""
+
+
+LICENSE_ICON_BASE_URI = '/icons/licenses'
+LICENSES = {
+    'http://arxiv.org/licenses/nonexclusive-distrib/1.0/': {
+        'uri': 'http://arxiv.org/licenses/nonexclusive-distrib/1.0/',
+        'label': 'arXiv.org perpetual, non-exclusive license to '
+        'distribute this article',
```
Comment (erickpeirson) [inline arxiv/license.py]: Preserve visual indentation?

```
        'label': 'arXiv.org perpetual, non-exclusive license to '
                 'distribute this article',
```

File: arxiv/license.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    },
+    'http://creativecommons.org/licenses/by/4.0/': {
+        'uri': 'http://creativecommons.org/licenses/by/4.0/',
+        'label': 'Creative Commons Attribution license',
+        'order': 5,
+        'is_current': True,
+        'is_valid': True,
+        'icon_uri': f'{LICENSE_ICON_BASE_URI}/by-4.0.png'},
+
+    'http://creativecommons.org/licenses/by-sa/4.0/': {
```
Comment (erickpeirson) [inline arxiv/license.py]: Not really a problem, but curious why the URIs are used as keys? Seems like this struct could also be a list of dicts?

Comment (erickpeirson) [reply arxiv/license.py]: In that case, you could just use list order as de-facto order -- would make it easier to change later, and then you wouldn't need the sort in ``CURRENT_LICENSE_URIS``?

File: arxiv/tests/test_license.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+"""Tests for arXiv taxonomy module."""
+from unittest import TestCase
+from arxiv.license import LICENSES, CURRENT_LICENSE_URIS
+
+
+class TestLicense(TestCase):
+    """Tests for the arXiv license definitions."""
+
+    def test_current_license_uris(self):
```
Comment (erickpeirson) [inline arxiv/tests/test_license.py]: This is basically a regression test? Will it need to be updated when the ordering changes? 

Might consider some other validation checks here, for future changes to the licenses. E.g. assert that all items have uris with truthy str values, assert that all items have descriptions, etc. Thinking about things that should be universally true of licenses in this module, and for which there is a non-negligible probability of bungling in the future.

File: arxiv/license.py (hunk 2 above)
Comment (mhl10) [reply arxiv/license.py]: Thanks, good eye.

File: arxiv/license.py (hunk 1 above)
Comment (mhl10) [reply arxiv/license.py]: Agree!

File: arxiv/tests/test_license.py (hunk 4 above)
Comment (mhl10) [reply arxiv/tests/test_license.py]: This is indeed a regression test (and port of an old test) and would have to be updated with ordering changes. It probably creates more annoyance than actual value, so I'd be fine with removing it.

I've added some validation checks.

**Pull request merged**